generators/entity-client/templates/vue/src/main/webapp/app/entities/entity.service.ts.ejs

```diff
@@ -16,51 +16,51 @@ const baseSearchApiUrl = '<%= baseApi %>_search/<%= entityApiUrl %>?query=';
 export default class <%= entityAngularName %>Service {
   <%_ if (searchEngine === 'elasticsearch') { _%>
   public search(query<% if (pagination !== 'no') { %>, paginationQuery<% } %>) : Promise<any> {
-    return new Promise<any>(resolve => {
+    return new Promise<any>((resolve, reject) => {
       axios.get(`${baseSearchApiUrl}${query}<% if (pagination !== 'no') { %>&${buildPaginationQueryOpts(paginationQuery)}<% } %>`).then(function (res) {
         resolve(<% if (pagination !== 'no') { %>res<% } else { %>res.data<% } %>);
-      });
+      }).catch(err => { reject(err); });
     });
   }
   <%_ } _%>
 
   public find(id: <% if (pkType === 'String'  || pkType === 'UUID') { %>string<% } else { %>number<% } %>) : Promise<I<%= entityAngularName %>> {
-    return new Promise<I<%= entityAngularName %>>(resolve => {
+    return new Promise<I<%= entityAngularName %>>((resolve, reject) => {
       axios.get(`${baseApiUrl}/${id}`).then(function (res) {
         resolve(res.data);
-      });
+      }).catch(err => { reject(err); });
     });
   }
 
   public retrieve(<% if (pagination !== 'no') { %>paginationQuery?: any<% } %>) : Promise<any> {
-    return new Promise<any>(resolve => {
+    return new Promise<any>((resolve, reject) => {
```

Comment:
For me, all this can be simplified by `return axios.get(baseApiUrl<% if (pagination !== 'no') { %> + ``?${buildPaginationQueryOpts(paginationQuery)}`` <% } %>)`